game/src/sandbox/dashboards/travel_times.rs

```diff
@@ -467,20 +502,30 @@ fn contingency_table(ctx: &mut EventCtx, app: &App, filter: &Filter) -> Widget {
             .secondary()
             .into_widget(ctx),
     ])
-    .padding(16)
-    .outline(ctx.style().section_outline)
 }
 
 pub struct Filter {
     changes_pct: Option<f64>,
     modes: BTreeSet<TripMode>,
+    include_no_changes: bool,
+}
+
+impl TripProblemFilter for Filter {
+    fn includes_mode(&self, mode: &TripMode) -> bool {
+        self.modes.contains(mode)
+    }
+
+    fn include_no_changes(&self) -> bool {
+        self.include_no_changes
+    }
 }
 
 impl Filter {
     pub fn new() -> Filter {
         Filter {
             changes_pct: None,
             modes: TripMode::all().into_iter().collect(),
+            include_no_changes: false,
```

Comment:
![screencast](https://user-images.githubusercontent.com/1664407/118344712-9d53f380-b4e4-11eb-97d7-cee64b9eb68c.gif)
Checking the box resets the whole panel, which resets the scrollbar. It'll take a small bit of refactoring, but at the end of handling `Outcome::Changed`, we could create the new panel (splitting out most of the logic from `new_state`), then do `new_panel.restore(ctx, &self.panel)` to preserve the scrollbar position. I can fix this up during the hackathon tomorrow if you're done for the night

Reply:
I fixed this (and also rebased on master to try to use the savestate + prebaked thing, which I'm now seeing still doesn't work...)